stingray/covariancespectrum.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import division
+
+import numpy as np
+
+from stingray import Lightcurve
+import stingray.utils as utils
+
+__all__ = ['Covariancespectrum']
+
+
+class Covariancespectrum(object):
+
+    def __init__(self, event_list, dt, band_interest=None,
+                 ref_band_interest=None):
+        """
+        Parameters
+        ----------
+        event_list : numpy 2D array
+            A numpy 2D array with first column as time of arrival and second
+            column as photon energies associated.
+
+        dt : float
+            The time resolution of the Lightcurve formed from the energy bin.
+
+        band_interest : iterable of tuples, default All
+            An iterable of tuples with minimum and maximum values of the range
+            in the band of interest. e.g list of tuples, tuple of tuples.
+
+        ref_band_interest : tuple of reference band range, default All
+            A tuple with minimum and maximum values of the range in the band
+            of interest in reference channel.
+
+        Attributes
+        ----------
+        energy_events : dictionary
+            A dictionary with energy bins as keys and time of arrivals of
+            photons with the same energy as value.
+
+        energy_covar : dictionary
+            A dictionary with mid point of band_interest and their covariance
+            computed with their individual reference band.
+
+        covar : np.ndarray
+            An array of arrays with mid point band_interest and their
+            covariance. It is the array-form of the dictionary `energy_covar`.
+
+        min_time : int
+            Time of arrival of the earliest photon.
+
+        max_time : int
+            Time of arrival of the last photon.
+
+        min_energy : float
+            Energy of the photon with the minimum energy.
+
+        max_energy : float
+            Energy of the photon with the maximum energy.
+        """
+        self.min_energy = np.min(event_list.T[1])
+        self.max_energy = np.max(event_list.T[1])
+        self.min_time = np.min(event_list.T[0])
+        self.max_time = np.max(event_list.T[0])
+
```

Comment:
It seems to me that this does the transpose four times when it really only needs to do it once? This might be slow for long event lists? I'd maybe pull line 89 (`event_list_T`) up here, or is that not possible?